.github/workflows/python_push_master.yml

```diff
@@ -0,0 +1,112 @@
+name: Python - Master Push
+
+# run (and build) only py3.10 but on all platforms
+# pull_request.yml tests all py versions but only on ubuntu
+
+on:
+  push:
+    # branches-ignore:
+    # # TODO: change this to `master`
+    #   - '*'
+    branches:
+      - 'nn-python-packaging'
+    paths-ignore:
+      - "*.md"
+      - .circleci/
+      - docs/
+      - run_route_scripts/
+      - test_requests/
+
+jobs:
+  build_wheels:
+    name: Push - cp*, ${{ matrix.platform }}
+    runs-on: ${{ matrix.image }}
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - image: ubuntu-latest
+            platform: linux
+          # - image: windows-latest
+          #   platform: windows
+          # - image: macos-latest
+          #   platform: macos
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        # we need to fetch all tags for setuptools-scm to work properly
+        fetch-depth: 0
+        fetch-tags: true
+
+    - name: Configure Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.12'
+
+    - name: Get branch name
+      run: echo "VALHALLA_BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_ENV
+
+    - name: Determine package to publish, pyvalhalla(-git)
+      run: |
+        VALHALLA_RELEASE_PKG="pyvalhalla"
+        if [[ "${GITHUB_REF#refs/heads/}" != "refs/tags*" ]]; then
+          VALHALLA_RELEASE_PKG="${VALHALLA_RELEASE_PKG}-git"
+        fi
+        echo "VALHALLA_RELEASE_PKG=${VALHALLA_RELEASE_PKG}" >> $GITHUB_ENV
+
+    - name: Restore cache
```

Comment:
cache isn't properly working yet.. currently the whole workflow is taking 15 mins, so it won't make us wait more than usual at least. I'd like to solve the caching problem, it's a bit annoying bcs ccache is runnnig inside the docker container..